src/vtt-segment-loader.js

```diff
@@ -154,6 +154,15 @@ export default class VTTSegmentLoader extends SegmentLoader {
    * @param {number} end - the end time of the region to remove from the buffer
    */
   remove(start, end) {
+    // if we only have "one" segment we likely have the entire caption file in a single
+    // and in that case we shouldn't remove cues because they won't get re-added later on.
+    // Also, if we're live, we probably want to remove them still as more segments could be added.
+    if (this.playlist_ &&
+      this.playlist_.endList &&
```

Comment:
Judging by the comment should this be `!this.playlist_.endList`?

Reply:
We want to skip removal of cues if we are VOD (`endList`) and we have 1 segment. This is an occasion where explicitly checking `endList === true` might be warranted.

Reply:
your right :doh: